### ISSUE TITLE
Character Counter/Limit

### ISSUE THREAD

**Issue by pimago** [tiff/wysihtml5#96]
Hi, i hope this is the place to post this.

Is there a character counter/limiter in wysihtml5 or are planning on adding one?

cheers,
Phil

**Comment by Petah** [comment]
On a side note the Raptor Editor has a character limit plugin, which you can see here http://www.raptor-editor.com/demo/docking

**Comment by pimago** [comment]
Yeah thanks. But raptor doesn't play nice with older browsers as wysihtml5 does …

**Comment by NWSE-ICO** [comment]
If there is a limitation in the backend, the user should see it in the frontend while she types. So I think a counter is a valid UX requirement for the common usecase of text lenght limitation. (What about a image count limitation then -- just kidding). 

But I'd leave further requirements on word count and statistics to MS Word. Please.

**Comment by Petah** [comment]
Yep, Raptor is only just released. Post your issues/feature requests and we will give you priority support. It works well in IE 9, what are your other target browsers?

**Comment by tiff** [comment]
@Petah can you please use a different way for promoting your editor. Raptor looks nice. I especially like the "raptorize" option. Thanks.

@pimago Having a character limit option could be neat. A good approach would be to copy over the maxlength attribute from the original textarea and simulate its behavior. But what about images and similar objects?

**Comment by Petah** [comment]
Sorry Tiff, will do :)

**Comment by pimago** [comment]
@tiff that's what I am trying right know. Actually with several editors, but I cannot mange to workt it out. Then I stumpled upon wysihtml5, I thought it could do it.

In my case I don't need an image limit. I will keep a look on wysihtml5 until it features a counter :)

**Comment by RoyLee** [comment]
I was able to implement this for the "composer" area but am having issues in the html "textarea".  What tag should I be looking at to count the value when I'm in the html "textarea"?  Here is the current code:

```
var wysihtml5_max_length = 1000;
var wysihtml5_cur_length = 0;

function verifyWysihtml5CurLength()
{
  wysihtml5_cur_length = document.getElementsByClassName('wysihtml5-sandbox')[0].contentWindow.document.body.innerHTML.length;

  if (wysihtml5_cur_length > wysihtml5_max_length)
    $("#wysihtml5_length_counter").html("<font style='color:red'>You've exceeded the maximum by " + (wysihtml5_cur_length - wysihtml5_max_length) + " characters.  Anything past the limit will not be saved.</font>");
  else
    $("#wysihtml5_length_counter").html((wysihtml5_max_length - wysihtml5_cur_length) + " characters left.");
}

editor.on("aftercommand:composer", verifyWysihtml5CurLength);
editor.on("change:composer", verifyWysihtml5CurLength); //Works
editor.on("change:textarea", verifyWysihtml5CurLength); //Does not work
```

**Comment by thuatnha** [comment]
How to add Code?
plz help me

**Comment by elsurudo** [comment]
Any word on this? Does anyone have a decent workaround?

**Comment by constantm** [comment]
@elsurudo this repo is not maintained any longer. I just opened an issue on the maintained version here: https://github.com/Voog/wysihtml/issues/149

**Comment by USA24** [comment]
http://www.twitter.com/usasunlight

**Comment by USA24** [comment]
<h1>WYSIHTML5 - A better approach to rich text editing</h1>
<p>wysihtml5 is an <span class="wysiwyg-color-green"><a href="https://github.com/xing/wysihtml5">open source</a></span> rich text editor based on HTML5 technology and the progressive-enhancement approach.
It uses a sophisticated security concept and aims to generate fully valid HTML5 markup by preventing unmaintainable tag soups and inline styles.</p>
<h2>Features</h2>
<ul>
  <li>It's fast and lightweight (smaller than TinyMCE, Aloha, ...)</li>
  <li>Auto-linking of urls as-you-type</li>
  <li>Generates valid and semantic HTML5 markup (even when the content is pasted from MS Word)</li>
  <li>Uses class names instead of inline styles</li>
  <li>Unifies line break handling across browsers</li>
  <li>Uses sandboxed iframes in order to prevent identity theft through XSS</li>
  <li>Speech-input for Chrome</li>
  <li>No jQuery required</li>
</ul>
<h2>Browser Support</h2>
<ul>
  <li><img src="http://xing.github.com/wysihtml5/img/icn_firefox.png" width="24" height="24"> Firefox 3.5+</li>
  <li><img src="http://xing.github.com/wysihtml5/img/icn_chrome.png" width="24" height="24"> Chrome</li>
  <li><img src="http://xing.github.com/wysihtml5/img/icn_internet_explorer.png" width="24" height="24"> IE 8+</li>
  <li><img src="http://xing.github.com/wysihtml5/img/icn_safari.png" width="24" height="24"> Safari 4+</li>
  <li><img src="http://xing.github.com/wysihtml5/img/icn_ios.png" width="24" height="24"> Safari on iOS 5+</li>
  <li><img src="http://xing.github.com/wysihtml5/img/icn_opera.png" width="24" height="24"> Opera 11+</li>
  <li><strong>Graceful degradation:</strong> Unsupported browsers will get a <i>&lt;textarea&gt;</i></li>
</ul>
